build/opts_templates/mongoc-opts.c.template

```diff
@@ -19,7 +19,7 @@ _{{ struct_name }}_parse (
 {
    bson_iter_t iter;
 
-   BSON_ASSERT_PARAM (client);
+   BSON_ASSERT (client || true); // client may be NULL.
```

Comment:
Is there any value to retaining this line given `|| true`? It might make more sense to replace this with a comment explaining why you don't assert `client` at all, and how it's not really necessary since the option validation itself doesn't work with a client and any called functions that do will perform a real assertion.

Up to you.

Reply:
I have seen the `|| true` pattern recently introduced as a way to document "this option has been considered, and is explicitly OK to be NULL". [Example](https://github.com/mongodb/mongo-c-driver/blob/6f58b9d7f32d5ea83b32c162a59d1f97e958657b/src/libmongoc/src/mongoc/mongoc-cluster.c#L2410-L2420).